### PR TITLE
Adapt Tracker Alignment PV Validation to work with phase-2 geometries

### PR DESCRIPTION
**Pull request by mmusich**
#### PR description:

The goal of this PR is to adapt the Tracker Alignment PV Validation tool to be able to cope with Phase-2 Tracker geometries.
The main things which are changed are:
   * probe track pseudo-rapidity acceptance is enlarged to cover up to  |η=4| (which is the maximal acceptance of TEPX)
   * the number of modules in one TBPX ladder along z is 9 and not 8 as for the phase-0/phase-1 detectors

Shown here for reference is the layout of the T15 (included in the D49) geometry, as used for the Phase-2 HLT TDR: 

image

#### PR validation:

Relies on the existing unit tests.

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

Not a backport, no backport is needed.

### CODE REVIEW

**Comment by mmusich** [comment]
@cmsbuild, please test

File: Alignment/OfflineValidation/plugins/PrimaryVertexValidation.cc [hunk 1]
```diff
@@ -1097,27 +1157,29 @@ void PrimaryVertexValidation::analyze(const edm::Event& iEvent, const edm::Event
...
+      } else if (thePhase != PVValHelper::phase2) {  // should be SiStrip now
         if (dynamic_cast<const SiStripRecHit2D*>(&hit))
           return false;  // normal hit
         else if (dynamic_cast<const SiStripMatchedRecHit2D*>(&hit))
           return true;  // matched is 2D
         else if (dynamic_cast<const ProjectedSiStripRecHit2D*>(&hit))
           return false;  // crazy hit...
         else {
-          edm::LogError("UnkownType") << "@SUB=AlignmentTrackSelector::isHit2D"
+          edm::LogError("UnkownType") << "@SUB=PrimaryVertexValidation::isHit2D"
```
Comment (christopheralanwest) [inline Alignment/OfflineValidation/plugins/PrimaryVertexValidation.cc:1176]: "Unknown" is misspelled. This misspelling is present in several places in CMSSW: https://github.com/cms-sw/cmssw/search?q=UnkownType I suggest changing the misspellings all at once but will sign off on the PR anyway. You can decide if you want to fix them now or later; I don't have an opinion.

**Comment by christopheralanwest** [comment]
+1

**Comment by qliphy** [comment]
+1

File: Alignment/OfflineValidation/plugins/PrimaryVertexValidation.cc (hunk 1 above)
Comment (mmusich) [reply Alignment/OfflineValidation/plugins/PrimaryVertexValidation.cc:1176]: @christopheralanwest 
it's addressed at https://github.com/cms-sw/cmssw/pull/32432

**Pull request merged**